pkg/expressions/json_templates_test.go

```diff
@@ -284,6 +284,77 @@ value13: | # This is a string
 				require.Equal(t, "42\n", parsed["value13"])
 			},
 		},
+		{
+			name:         "missing closing tag",
+			jsonTemplate: `{ "AString": "${{ aString " }`,
+			assertions: func(t *testing.T, _ []byte, err error) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "unclosed expression")
+			},
+		},
+		{
+			name:         "closing delimiter in expression body terminates early",
+			jsonTemplate: `{ "AString": "${{ '{{.domain}}' }}" }`,
+			assertions: func(t *testing.T, _ []byte, err error) {
+				// This fails because }} appears in the expression body
+				// The parser finds the first }} and terminates, leaving unterminated string
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "literal not terminated")
+			},
+		},
+		{
+			name:         "different delimiter type allows problematic characters",
+			jsonTemplate: `{ "AString": "${% '{{.domain}}' %}" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := map[string]any{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				require.Equal(t, "{{.domain}}", parsed["AString"])
+			},
+		},
+		{
+			name:         "delimiter starts can appear as literals in expressions",
+			jsonTemplate: `{ "AString": "${{ 'This string contains literal ${% inside it' }}" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := testStruct{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				require.Equal(t, "This string contains literal ${% inside it", parsed.AString)
+			},
+		},
+		{
+			name:         "multiple delimiters can coexist in the same template",
+			jsonTemplate: `{ "AString": "foo ${{ aString }} and ${% 'bar' %}" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := testStruct{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				// Both expressions are evaluated
+				require.Equal(t, "foo hello and bar", parsed.AString)
+			},
+		},
+		{
+			name:         "nested delimiter start inside expression - final closer is literal",
+			jsonTemplate: `{ "AString": "${{ 'text ${%' }}%}" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := testStruct{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				// ${{ expr }} is evaluated, then "%}" remains as literal text
+				require.Equal(t, "text ${%%}", parsed.AString)
+			},
+		},
```

Comment:
This appears to test the same thing as the [other](https://github.com/akuity/kargo/pull/5473/files#r2620359471) test cases I suggested removing. This should go, too.

```suggestion
```